extensions/sql/contract-definition/store/src/main/java/org/eclipse/dataspaceconnector/sql/contractdefinition/store/SqlContractDefinitionStore.java

```diff
@@ -133,11 +93,11 @@ public void save(Collection<ContractDefinition> definitions) {
         transactionContext.execute(() -> {
             try (var connection = getConnection()) {
                 for (var definition : definitions) {
-                    executeQuery(connection, SQL_SAVE_CLAUSE_TEMPLATE,
+                    executeQuery(connection, statements.getInsertTemplate(),
```

Comment:
You should check beforehand with "existById(definition.getId, connection)" if a ContractDefinition with same ID exist before you insert it.

Reply:
The `ContractDefinitionStore` has an explicit `update` method, so we don't need `upsert` semantics here.
Removing the `update` method altogether, and implementing `upsert` semantics is possible, but something for another PR, as it will affect a rather wide swath of files.

Reply:
on closer inspection, I noticed that the `Cosmos` impl also has upsert semantics. Will make it consistent then.